src/int_id.h

```diff
@@ -128,4 +128,37 @@ struct hash< int_id<T> > {
 };
 } // namespace std
 
+/**
+ * A tiny fixed hashtable-based cache that speeds up `int_id::obj()` calls.
+ * Exploits the following facts:
+ * 1. same ids are usually clustered together (i.e. fields, furn, ter)
+ * 2. cache is tiny -> less CPU cache misses
+ * 3. if "obj" was stored in the cache, it's guaranteed to be valid
+ * Intended to be used locally in situations when `int_id::obj()` is called in a tight loop.
+ * Create the cache before the loop and then instead of `id.obj()` call `cache.obj(id)`.
+ * @tparam T type of the object of the corresponding int_id
+ */
+template<typename T>
+class int_id_cache
```

Comment:
To answer your question: I would put this class in a new header of its own.  `int_id.h` is included a lot, so we should try to keep it lightweight.

Adding new headers is usually not an issue for compile times; adding new cpp files is what causes problems.  And luckily this code is header-only.